baremaps-core/src/main/java/org/apache/baremaps/workflow/tasks/CreateIplocIndex.java

```diff
@@ -26,101 +26,93 @@
 import java.io.BufferedInputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.UncheckedIOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Objects;
-import java.util.stream.Stream;
-
-public record CreateIplocIndex(String geonamesIndexPath, String[] iplocNicPath, String targetIplocIndexPath) implements Task {
-
-  private static final Logger logger = LoggerFactory.getLogger(CreateIplocIndex.class);
-
-  @Override
-  public void execute(WorkflowContext context) throws Exception {
-    logger.info("Generating Iploc from {} {}", geonamesIndexPath, iplocNicPath);
-
-    logger.info("Creating the Geocoder");
-    GeonamesGeocoder geocoder;
-    try {
-      geocoder = new GeonamesGeocoder(Path.of(geonamesIndexPath), null);
-      if (!geocoder.indexExists()) {
-        logger.error("Geocoder index doesn't exist");
-        return;
-      }
-      geocoder.open();
-    } catch(Exception e) {
-      logger.error("Error while creating the geocoder index", e);
-      return;
-    }
 
-    logger.info("Generating NIC objects stream");
-    Stream<NicObject> fetchNicObjectStream = Arrays.stream(iplocNicPath).flatMap(iplocNicPath -> {
-      try {
-        InputStream inputStream = new BufferedInputStream(Files.newInputStream(Path.of(iplocNicPath)));
-        return NicParser.parse(inputStream).onClose(() -> {
-          try {
-            inputStream.close();
+public record CreateIplocIndex(String geonamesIndexPath, String[] nicPaths,
+                               String targetIplocIndexPath) implements Task {
+
+    private static final Logger logger = LoggerFactory.getLogger(CreateIplocIndex.class);
+
+    @Override
+    public void execute(WorkflowContext context) throws Exception {
+        logger.info("Generating Iploc from {} {}", geonamesIndexPath, nicPaths);
+
+        logger.info("Creating the Geocoder");
+        GeonamesGeocoder geocoder;
+        try {
+            geocoder = new GeonamesGeocoder(Path.of(geonamesIndexPath), null);
+            if (!geocoder.indexExists()) {
```

Comment:
In an other PR, we may think about improving error handling. I wonder if we should throw an exception and handle/log the error at the level of the workflow. What do you think?

Reply:
I think we can go either way. What I would do is return a flag that tells the Workflow if the task has succeeded or not, if it has failed the workflow should abort. However I think that exceptions should be caught by the task.